### PR TITLE
wxwidgets: fix building 3.1 on darwin

### PR DESCRIPTION
**Pull request by knl**
###### Description of changes

Fix build failures on darwin by passing all the necessary dependencies and removing deprecated command line switches to configure.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tfmoraes** [comment]
@knl there is already a commit in Nixpkg to update wxwidgets https://github.com/NixOS/nixpkgs/pull/157848

**Comment by knl** [comment]
@tfmoraes thanks, haven’t seen it, as I was not looking in staging. Once it reaches master, maybe this PR could be useful, since it fixes building on darwin?

**Comment by knl** [comment]
@tfmoraes I updated the PR to make 3.1 build on darwin. Please take a look. I'm wondering should we carve out a dedicated version without gtk, just for mac?

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -20682,7 +20682,10 @@ with pkgs;
 
   wxmac = callPackage ../development/libraries/wxwidgets/wxmac30.nix { };
 
-  wxGTK31 = callPackage ../development/libraries/wxwidgets/wxGTK31.nix { };
+  wxGTK31 = callPackage ../development/libraries/wxwidgets/wxGTK31.nix {
+    withGtk2 = true;
```
Comment (tfmoraes) [inline pkgs/top-level/all-packages.nix:20686]: it should be withGTK2 = false,  the idea is to compile with GTK 3.

Comment (knl) [reply pkgs/top-level/all-packages.nix:20686]: The default in `wxGTK31.nix` is `withGtk2 ? true`. I changed it so that on darwin it's gtk3.

**Review by tfmoraes — APPROVED** [review]

File: pkgs/development/libraries/wxwidgets/wxGTK31.nix [hunk 2]
```diff
@@ -17,20 +17,21 @@
...
 , withMesa ? lib.elem stdenv.hostPlatform.system lib.platforms.mesaPlatforms
-, withWebKit ? false, webkitgtk
+, withWebKit ? stdenv.isDarwin
+, webkitgtk
 , darwin
 }:
 
 assert withMesa -> libGLU != null && libGL != null;
-assert withWebKit -> webkitgtk != null;
+assert withWebKit -> webkitgtk != null || stdenv.isDarwin;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/wxwidgets/wxGTK31.nix:28]:
```suggestion
assert withWebKit -> stdenv.isDarwin;
```

File: pkgs/development/libraries/wxwidgets/wxGTK31.nix [hunk 3]
```diff
@@ -17,20 +17,21 @@
...
-, withGtk2 ? true
+, withGtk2 ? (!stdenv.isDarwin)
 , withMesa ? lib.elem stdenv.hostPlatform.system lib.platforms.mesaPlatforms
-, withWebKit ? false, webkitgtk
+, withWebKit ? stdenv.isDarwin
+, webkitgtk
 , darwin
 }:
 
 assert withMesa -> libGLU != null && libGL != null;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/wxwidgets/wxGTK31.nix:27]:
```suggestion
```

File: pkgs/development/libraries/wxwidgets/wxGTK31.nix [hunk 4]
```diff
@@ -17,20 +17,21 @@
 , compat28 ? false
 , compat30 ? true
 , unicode ? true
-, withGtk2 ? true
+, withGtk2 ? (!stdenv.isDarwin)
 , withMesa ? lib.elem stdenv.hostPlatform.system lib.platforms.mesaPlatforms
-, withWebKit ? false, webkitgtk
+, withWebKit ? stdenv.isDarwin
+, webkitgtk
 , darwin
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/wxwidgets/wxGTK31.nix:24]:
```suggestion
, 
```
Please input the frameworks individually.

Comment (knl) [reply pkgs/development/libraries/wxwidgets/wxGTK31.nix:24]: Gladly! I just took the existing style and was wondering why it's only having darwin.

**Pull request merged**